### ISSUE TITLE
Azure functions proxies changes timezone offset for values hardcoded in responseOverrides to the local timezone of the host.

### ISSUE THREAD

**Issue by Vidarls** [Azure/azure-functions-host#7426]
I have set up a simple proxy for test use that should return some static json data when responding to a request.

Part of this data is date / time values with a specific timezone offset. 

When the proxy responds all the date/time values are altered to be in the timezone of the host. (Locally it gets my current timezone, in Azure it becomes UTC).

#### Investigative information

This setup in proxies.json:

```
(...)
"responseOverrides": {
    "response.body": {
        "status": 1,
        "message": "Success",
        "data": {
            "ID": [
                {
                    "start_time": "2021-05-20T00:30:00-05:00",
                    "end_time": "2021-05-20T01:54:00-05:00"
                } (...)
```

Yields the following when requested through the proxy:

```
HTTP/1.1 200 OK
Connection: close
Date: Wed, 02 Jun 2021 14:12:46 GMT
Content-Type: application/json
Server: Kestrel
Content-Length: 6689

{
  "status": 1,
  "message": "Success",
  "data": {
    "ID": [
      {
        "start_time": "2021-05-20T07:30:00+02:00",
        "end_time": "2021-05-20T08:54:00+02:00"
      } (...)
```

#### Repro steps

1. Set up a proxy with some hardcoded response body override that includes a datetime in iso format with timezone offset specified (must be different from the timezone of the host).
2. Call the proxy endpoint

#### Expected behavior

The content provided as a hardcoded value in `proxies.json` should be returned verbatim

#### Actual behavior

Datetime values are returned converted to the timezone offset of the current host.

I have observed this both locally using Azure core tools, and when hosted in Azure.

#### Known workarounds

None

#### Related information 

Provide any related information 

* Language of function app: Python (Not sure if relevant to proxies)
* Links to source: N/A
* Bindings used: N/A
* Version: Not sure about the Azure hosted one, locally I use core tool v `3.0.3568`

**Comment by v-anvari** [comment]
Hi @Vidarls , Thank you for your feedback! We will investigate this further and update you with the findings.

**Comment by v-anvari** [comment]
Tagging @galiniliev / @safihamid / for more information, Is this supported while using proxies?

**Comment by safihamid** [comment]
Yes it uses local timezone of the host, fixing it could cause breaking change for other customers.
here is a simple repro:

{
    "$schema": "http://json.schemastore.org/proxies",
    "proxies": {
        "timestamptest": {
            "matchCondition": {
                "route": "/time"
            },
            "responseOverrides": {
                "response.body": "{\r\n\"t1\": \"**2021-05-20T00:30:00-03:00**\",\r\n\"t2\": \"**2021-05-20T00:30:00**\"\r\n}"
            }
        }
    }
}

and here is the output:

{"t1":"**2021-05-20T03:30:00+00:00**","t2":"**2021-05-20T00:30:00**"}

note t1 value is updated because it has 2021-05-20T00:30:00**-03:00** but the value for t2 is returned 'as is'

so one possible workaround could be to have hardcoded dates/times and not with date/time additions/subtractions

**Comment by Vidarls** [comment]
Hi, thanks for looking into this.

I want to comment on the remark by @safihamid :

> so one possible workaround could be to have hardcoded dates/times and not with date/time additions/subtractions

 A date / time like `2021-06-15T16:45:33-05:00` is not an arithmetic operation. It is the ISO 8601 standard of formatting a date / time string. (https://en.m.wikipedia.org/wiki/ISO_8601). And the subtraction or addition at the end notes the timezone offset from UTC time. It does not indicate an actual subtraction or addition to the given time.

JSON itself does *not* have a supported date / time type, but recommend using a *string* representation in the ISO 8601 format (https://docs.jsonata.org/date-time).

I as a developer am a bit ambushed by the runtime modifying my hardcoded strings just because they happen to parse as date/time values. This is not something I would expect the framework to do.

Secondly, by doing this conversion the platform is losing some of my data. I have no way of knowing the date part of the original value, because I have no way of reversing the process without having some record of the source offset.

I understand that other users might now be relying on the existing behaviour, which makes this whole ordeal much more complicated to manage. 

Yet, if possible, I would still like to see this part of functions behaving in a predictable manner that does not suprise developers by altering hardcoded data points and cause loss of data.

**Comment by Vidarls** [comment]
I see that my link to show how json recommends using RFC 8601 is not to json.org, but rather to some programming language interpretation of how json works. My apologies for the misleading link.

I did find this from https://www.json.org/json-en.html :

> A value can be a string in double quotes, or a number, or true or false or null, or an object or an array. These structures can be nested.

Confirming that my datetime values should be treated as strings.

The closest thing I can find to a formalized standard for json mentioning how to handle date and time values is this: https://datatracker.ietf.org/doc/html/rfc7493#section-4.3.

**Comment by safihamid** [comment]
Thanks @Vidarls. I verified this is working fine in Azure API Management Service.
@galiniliev can you or someone from your team see what's the difference here please?
for APIM this works fine:
 <set-body template="none">{"t1": "2021-05-20T00:30:00-03:00","t2": "2021-05-20T00:30:00"}</set-body>

but in Proxies the value of t1 is updated based on utc time:
 "responseOverrides": {
"response.body": "{\r\n"t1": "2021-05-20T00:30:00-03:00",\r\n"t2": "2021-05-20T00:30:00"\r\n}"
}

The Proxy code which changes this is in: \src\Microsoft.Azure.AppService.Proxy.Common\Utility\Utility.cs

        public static bool TryParseJson(string json, out JToken root)
        {
            //Quick check if the given string is potentially json.
            //This is to avoid expensive parsing/exception in most cases
            if (string.IsNullOrWhiteSpace(json) ||
                (!json.StartsWith("{") &&
                 !json.StartsWith("[")))
            {
                root = null;
                return false;
            }

            try
            {
                **root = JToken.Parse(json);**
            }
            catch (JsonReaderException)
            {
                root = null;
                return false;
            }
            return true;
        }

JToken.Parse line changes the value.
